vnext/JSI.Desktop.UnitTests/JSI.Desktop.UnitTests.vcxproj

```diff
@@ -0,0 +1,145 @@
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <!-- We need to figure out how we want to consume the Google Test Nuget for
+    Win32. For now, we disable the Win32. -->
+    <!-- <ProjectConfiguration Include="Debug|Win32">
+      <Configuration>Debug</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration> -->
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{6f354505-fe3a-4bd2-a9a6-d12bbf37a85c}</ProjectGuid>
+    <ProjectName>JSI.Desktop.UnitTests</ProjectName>
+  </PropertyGroup>
+  <PropertyGroup Label="NOJSC">
+    <NOJSC>true</NOJSC>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <Import Project="$(ReactNativeWindowsDir)PropertySheets\React.Cpp.props" />
+  <PropertyGroup Label="Configuration">
+    <ConfigurationType>Application</ConfigurationType>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
+  <!-- We cannot include Warnings.props here because Google Test only builds at warning level 3. -->
+  <!-- <Import Project="$(ReactNativeWindowsDir)PropertySheets\Warnings.props" /> -->
+  <ImportGroup Label="ExtensionSettings">
+  </ImportGroup>
+  <ImportGroup Label="Shared">
+    <Import Project="..\Chakra\Chakra.vcxitems" Label="Shared" />
+    <Import Project="..\Shared\Shared.vcxitems" Label="Shared" />
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup>
+    <IncludePath>$(ReactNativeWindowsDir);$(ReactNativeWindowsDir)stubs;$(FollyDir);$(ReactNativeDir)\ReactCommon;$(ReactNativeWindowsDir)ReactWindowsCore;$(ReactNativeWindowsDir)include\ReactWindowsCore;$(ReactNativeWindowsDir)Desktop;$(ReactNativeWindowsDir)IntegrationTests;$(MSBuildProjectDirectory);$(IncludePath)</IncludePath>
+    <LinkIncremental>true</LinkIncremental>
+  </PropertyGroup>
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <!--
+        BOOST_ASIO_HAS_IOCP - Force unique layout/size for boost::asio::basic_stream_socket<> subtypes.
+        REACTWINDOWS_BUILD - building with REACTWINDOWS_API as dll exports
+      -->
+      <PreprocessorDefinitions>REACTWINDOWS_BUILD;BOOST_ASIO_HAS_IOCP;_WINSOCK_DEPRECATED_NO_WARNINGS;_WIN32_WINNT=_WIN32_WINNT_WIN7;WIN32;_WINDOWS;_USRDLL;REACTNATIVEWIN32_EXPORTS;FOLLY_NO_CONFIG;NOMINMAX;GLOG_NO_ABBREVIATED_SEVERITIES;_HAS_AUTO_PTR_ETC;CHAKRACORE;RN_EXPORT=;NOJSC;GTEST_LANG_CXX11;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

Comment:
>BOOST_ASIO_HAS_IOCP [](start = 50, length = 19)

(nit) this line is super long, can we split it into multiple lines?

Reply:
@JunielKatarn, do you know if we need to keep this line as such for compatibility with the Visual Studio UI?

Reply:
Look into `PropertySheets`. Some of these macros may be redundant.
I believe you ONLY need to keep `GTEST_LANG_CXX11`.
`NOJSC` is effectively deprecated.

Reply:
new comment

---
In reply to: [298655081](https://github.com/microsoft/react-native-windows/pull/2677#discussion_r298655081) [](ancestors = 298655081)

Reply:
I removed NOJSC and _USRDLL. I don't think the rest of the macros are defined in React.Cpp.Props, and many of them are used in almost all vcxprojs. Let's clean this up in a separate PR.

---
In reply to: [298775253](https://github.com/microsoft/react-native-windows/pull/2677#discussion_r298775253) [](ancestors = 298775253)